src/xact.cc

```diff
@@ -211,13 +224,22 @@ bool xact_base_t::finalize()
   // If there is only one post, balance against the default account if one has
   // been set.
 
-  if (journal && journal->bucket && posts.size() == 1 && ! balance.is_null()) {
-    null_post = new post_t(journal->bucket, ITEM_INFERRED);
-    null_post->_state = (*posts.begin())->_state;
-    add_post(null_post);
+  if (journal && journal->bucket && posts.size() == 1) {
+    if (! balance_real.is_null()) {
+      null_post_real = new post_t(journal->bucket, ITEM_INFERRED);
+      null_post_real->_state = (*posts.begin())->_state;
+      add_post(null_post_real);
+    }
+    // TODO: does this need to set POST_VIRTUAL | POST_MUST_BALANCE ?
+    else if (! balance_virtual.is_null()) {
+      null_post_virtual = new post_t(journal->bucket, ITEM_INFERRED);
+      null_post_virtual->_state = (*posts.begin())->_state;
+      add_post(null_post_virtual);
+    }
   }
 
-  if (! null_post && balance.is_balance() &&
+  // TODO: How do we want handle virtual postings here?
+  if (! null_post_real && ! null_post_virtual && balance.is_balance() &&
       balance.as_balance().amounts.size() == 2) {
     // When an xact involves two different commodities (regardless of how
     // many posts there are) determine the conversion ratio by dividing the
```

Comment:
When there are two commodities, we calculate the conversion rate between them.

If there are two real and two virtual, should we do the same? Regardless of whether that's two, three, or four in total?